src/test/java/org/eclipse/uprotocol/testues/mock/MockUService.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.eclipse.uprotocol.testues.mock;
+
+import java.io.IOException;
+import java.net.UnknownHostException;
+
+import org.eclipse.uprotocol.cloudevent.datamodel.UCloudEventAttributes;
+import org.eclipse.uprotocol.cloudevent.factory.CloudEventFactory;
+import org.eclipse.uprotocol.cloudevent.factory.UCloudEvent;
+import org.eclipse.uprotocol.status.datamodel.UStatus;
+import org.eclipse.uprotocol.status.factory.UStatusFactory;
+import org.eclipse.uprotocol.ubus.EventListener;
+import org.eclipse.uprotocol.ubus.mock.MockUBus;
+import org.eclipse.uprotocol.uri.datamodel.UAuthority;
+import org.eclipse.uprotocol.uri.datamodel.UEntity;
+import org.eclipse.uprotocol.uri.datamodel.UResource;
+import org.eclipse.uprotocol.uri.datamodel.UUri;
+import org.eclipse.uprotocol.uri.factory.UriFactory;
+
+import com.google.protobuf.Any;
+
+import io.cloudevents.CloudEvent;
+
+public class MockUService implements EventListener, Runnable {
+
+    private MockUBus uBus;
+    
+    private final UEntity uEName = new UEntity("HartleyService", "1.0");
+
+    @Override
+    public UStatus onEvent(CloudEvent ce) {
+        // This is where we implement the request then generate a reply
+        System.out.println("Received Request " + ce.toString());
+
+        if ( (uBus != null) && (ce != null) && (!UCloudEvent.isExpired(ce)) &&
```

Comment:
(uBus != null) should be checked in the constructor when the service is built - not every time we get an event